dandiapi/api/migrations/0006_remove_embargoedassetblob_dandiset_and_more.py

```diff
@@ -0,0 +1,91 @@
+# Generated by Django 4.1.13 on 2024-01-16 18:31
+from __future__ import annotations
+
+from django.db import migrations, models
+
+
+def migrate_embargoed_asset_blobs(apps, _):
+    Asset = apps.get_model('api.Asset')
+    AssetBlob = apps.get_model('api.AssetBlob')
+
+    # We must include a filter to only include assets which are part of a version, as otherwise
+    # we may include an embargoed asset which has been previously updated. Since updating an asset
+    # results in a new asset which points to the same blob as the original asset, this would copy
+    # the same blob twice, resulting in an integrity error (same blob_id).
+    embargoed_assets = Asset.objects.filter(
+        embargoed_blob__isnull=False, versions__isnull=False
+    ).select_related('embargoed_blob')
+
+    # For each relevant asset, create a new asset blob with embargoed=True,
+    # and point the asset to that
+    for asset in embargoed_assets.iterator():
+        blob_id = str(asset.embargoed_blob.blob_id)
+        new_blob_location = f'blobs/{blob_id[0:3]}/{blob_id[3:6]}/{blob_id}'
+        new_asset_blob = AssetBlob.objects.create(
+            embargoed=True,
+            blob=new_blob_location,
+            blob_id=asset.embargoed_blob.blob_id,
+            created=asset.embargoed_blob.created,
+            modified=asset.embargoed_blob.modified,
+            sha256=asset.embargoed_blob.sha256,
+            etag=asset.embargoed_blob.etag,
+            size=asset.embargoed_blob.size,
+            download_count=asset.embargoed_blob.download_count,
+        )
+        asset.blob = new_asset_blob
+        asset.embargoed_blob = None
+        asset.save()
```

Comment:
I was originally going to include that as part of this migration, but after thinking about it more, I decided that it wouldn't accomplish anything. The reason being is that if an embargoed upload is in progress, when we migrate all of the data to the sponsored bucket, that in progress upload wouldn't be migrated to the new bucket. So attempting to finish that upload would fail since it wouldn't be able to find that s3 object in the sponsored bucket.

I figured the best approach was just to announce beforehand to everyone with embargoed dandisets to finish any current uploading by a specific date, and once the migration is complete, they can continue on afterwards as usual.